evaluation/generation/generate.py

```diff
@@ -0,0 +1,67 @@
+import argparse
+import json
+
+from transformers.models.bigscience176b import BigScience176BLMHeadModel
+from transformers import AutoTokenizer, AutoConfig, logging
+
+logging.set_verbosity_debug()
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    # --checkpoint /gpfswork/rech/six/uan68tv/model-conversion/main-gs-47400-transformers-sharded
+    # --checkpoint /gpfswork/rech/six/uan68tv/model-conversion/tr11e-350M-transformers-555750
+    parser.add_argument("--checkpoint", type=str, help="Checkpoint path")
+    parser.add_argument("--parallelize-for-176b", action="store_true")
+    parser.add_argument("--generate-max-length", type=int, default=50, help="max generation length")
+    parser.add_argument("--greedy", action="store_true")
+
+    return parser.parse_args()
+
+def generate_from_text(model, text, tokenizer, max_length=200, greedy=False):
+    input_ids = tokenizer.encode(text, return_tensors='pt').to("cuda:0")
+    max_length = input_ids.size(-1) + max_length
+    if greedy:
+        greedy_output = model.generate(input_ids.to('cuda:0'), max_length=max_length)
+    else:
+        greedy_output = model.generate(input_ids.to('cuda:0'), max_length=max_length, do_sample=True, top_k=0)
+    return {
+        "inputs": text,
+        "outputs": tokenizer.decode(greedy_output[0], skip_special_tokens=True)
```

Comment:
Maybe we can see if we can get only the "output" in the sense that current version will also contain the prompt.